docs/proposals/kep-003-service-resouce-aggregation.md

```diff
@@ -0,0 +1,57 @@
+# [KEP-003] Service resource aggregation
+
+## Background
+
+Currently, we're creating catapult deployment per each `CustromResourceDiscovery` object. The catapult component transports custom resources from the service cluster to the management cluster, and vice versa, asynchronously. For example the `db.example.io` apiGroup in the service cluster `svc-1` becomes `db.internal.svc-1.provider-1` apiGroup in the management cluster.
+
+Furthermore, catapult component handles the webhooks for internal apiGroup. If there is `CustromResourceDiscoverySet` there's a new catapult deployment per each service cluster offering the same service.
+
+* See also:
+
+https://github.com/kubermatic/kubecarrier/issues/316
+
+## Issues
+
+The issues with this approach are multi-fold:
+
+* propagation is asynchronously, whereas management clusters serves as a "cache" to service cluster's ground truth. In more complicated setups the management cluster's resource version and service cluster resource version could diverge in unreconciable way. To give a simple, albeit toy example consider the following:
+
+    * There's deployment on the service cluster. This resource is propagated to the management cluster. At the start the deployment replicas is 1.
+       * HPA controller on the service cluster increases the replica count to 2
```

Comment:
What is HPA controller?

Reply:
Who increases the replica count? The service provider modify the object in the service cluster directly or?

Reply:
HorizontalPodAutoscaler  

https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/